src/models.ts

```diff
@@ -297,3 +304,33 @@ export class AdvancedFilter extends Filter {
     return filter;
   }
 }
+
+export interface IDataReference {
+  target: IFilterTarget;
+}
+
+export interface IEqualsDataReference extends IDataReference {
+  equals: string | boolean | number;
+}
+
+export interface IBetweenDataReference extends IDataReference {
+  between: (string | boolean | number)[];
+}
+
+export interface IValueDataReference extends IDataReference {
+  value: string | boolean | number;
```

Comment:
Yes, I think it might raise questions why there's a value if it's always undefined. We could probably explain that the value is coming soon (hopefully).
Or we could have it initially optional and even keep it optional later not to introduce breaking changes.